### PR TITLE
Feature 489 requests identify as open mastr

### LINKED ISSUE #489: Identify as open-mastr package when requesting data from marktstammdatenregister.de
We could add a small identification in the request header to identify as python package user when the data is requested from BNetzA.

### PR DESCRIPTION
**Pull request by FlorianK13**
### Automation
Closes #489 

### PR-Assignee
- [ ] 🐙 Follow the workflow in [CONTRIBUTING.md](https://github.com/OpenEnergyPlatform/open-MaStR/blob/production/CONTRIBUTING.md)
- [ ] 📝 Update the [CHANGELOG.md](https://github.com/OpenEnergyPlatform/open-MaStR/blob/production/CHANGELOG.md)
- [ ] 📙 Update the documentation

### Reviewer
- [ ] 🐙 Follow the [Reviewer Guidelines](https://github.com/OpenEnergyPlatform/open-MaStR/blob/production/CONTRIBUTING.md#40-let-someone-else-review-your-pr)
- [ ] 🐙 Provided feedback and show sufficient appreciation for the work done

### CODE REVIEW

File: open_mastr/xml_download/utils_download_bulk.py [hunk 1]
```diff
@@ -117,20 +119,27 @@ def download_xml_Mastr(
     url = gen_url(now)
 
     time_a = time.perf_counter()
-    r = requests.get(url, stream=True)
+
+    headers = {"User-Agent": "python-package-open-mastr"}
```
Comment (chrwm) [inline open_mastr/xml_download/utils_download_bulk.py]:
```suggestion
    HEADERS = {"User-Agent": "python-package-open-mastr"}
```

File: open_mastr/xml_download/utils_download_bulk.py [hunk 2]
```diff
@@ -117,20 +119,27 @@ def download_xml_Mastr(
...
+    r = requests.get(url, stream=True, headers=headers)
```
Comment (chrwm) [inline open_mastr/xml_download/utils_download_bulk.py]:
```suggestion
    r = requests.get(url, stream=True, headers=HEADERS)
```

File: open_mastr/xml_download/utils_download_bulk.py [hunk 3]
```diff
@@ -117,20 +119,27 @@ def download_xml_Mastr(
...
-        log.warning("Download file was not found. Assuming that the new file was not published yet and retrying with yesterday.")
-        now = time.localtime(time.mktime(now) - (24 * 60 * 60)) # subtract 1 day from the date
-        r = requests.get(url, stream=True)
+        log.warning(
+            "Download file was not found. Assuming that the new file was not published yet and retrying with yesterday."
+        )
+        now = time.localtime(
+            time.mktime(now) - (24 * 60 * 60)
+        )  # subtract 1 day from the date
+        r = requests.get(url, stream=True, headers=headers)
```
Comment (chrwm) [inline open_mastr/xml_download/utils_download_bulk.py]:
```suggestion
        r = requests.get(url, stream=True, headers=HEADERS)
```

**Review by chrwm — CHANGES_REQUESTED** [review]
From what I understood, HEADERS is a constant that is not changed by the programme code. If so, I suggest to capitalise it for better readability and maintainability.

**Review by chrwm — DISMISSED** [review]
Thanks for the implementation!

**Review by chrwm — APPROVED** [review]

**Comment by FlorianK13** [comment]
This is paused, as a better solution exists in #526

**Pull request closed without merging**